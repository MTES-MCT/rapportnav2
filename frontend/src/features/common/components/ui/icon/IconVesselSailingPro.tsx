import Icon from '../../../../../assets/images/navire_plaisance_pro.svg'

function IconVesselSailingPro() {
  return <img src={Icon} alt="Navire de plaisance professionnelle" height={25} />
}

export default IconVesselSailingPro
