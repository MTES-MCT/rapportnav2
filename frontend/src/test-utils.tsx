import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
