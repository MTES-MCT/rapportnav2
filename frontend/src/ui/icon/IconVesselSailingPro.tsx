import Icon from '../../assets/images/navire_plaisance_pro.svg'

function IconVesselSailingPro() {
  return <img src={Icon} alt="Navire de commerce" />
}

export default IconVesselSailingPro
