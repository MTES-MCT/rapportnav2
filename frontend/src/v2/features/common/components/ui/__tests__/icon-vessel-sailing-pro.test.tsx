import { render } from '../../../../../../test-utils'
import IconVesselSailingPro from '../icon-vessel-sailing-pro'

describe('IconVesselSailingPro', () => {
  it('should match the snapshot', () => {
    const wrapper = render(<IconVesselSailingPro />)
    expect(wrapper).toMatchSnapshot()
  })
})
