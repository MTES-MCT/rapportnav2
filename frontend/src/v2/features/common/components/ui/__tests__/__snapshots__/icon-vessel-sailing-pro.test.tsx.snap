// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconVesselSailingPro > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="Navire de plaisance pro"
        src="data:image/svg+xml,%3csvg%20height='25'%20viewBox='0%200%2069%2025'%20width='69'%20xmlns='http://www.w3.org/2000/svg'%3e%3cclipPath%20id='a'%3e%3cpath%20d='m0%200h69v25h-69z'/%3e%3c/clipPath%3e%3cg%20clip-path='url(%23a)'%3e%3cpath%20d='m0%200h69v25h-69z'%20fill='transparent'/%3e%3cg%20fill='%233b4559'%3e%3cpath%20d='m542.789%202.775-2.524-2.525h-2.484l1.262%202.525zm0%200'%20transform='translate(-503.203%20-.25)'/%3e%3cpath%20d='m200.4.25h-5.68v2.525h7.58zm0%200'%20transform='translate(-182.207%20-.25)'/%3e%3cpath%20d='m0%20353.267h61.946l1.752-2.525h-63.698zm0%200'%20transform='translate(0%20-328.267)'/%3e%3cpath%20d='m61.385%20233.918h-61.385v5.048h65.492l3.508-5.048zm0%200'%20transform='translate(0%20-218.954)'/%3e%3cpath%20d='m110.971%2078.137h-45.2l-1.888%207.573h54.662zm-41.131%205.049h-2.525v-2.525h2.525zm5.049%200h-2.525v-2.525h2.524zm5.049%200h-2.525v-2.525h2.524zm5.05%200h-2.526v-2.525h2.525zm5.049%200h-2.526v-2.525h2.525zm5.049%200h-2.526v-2.525h2.524zm5.05%200h-2.527v-2.525h2.525zm5.048%200h-2.525v-2.525h2.525zm5.049%200h-2.525v-2.525h2.525zm0%200'%20transform='translate(-59.742%20-73.158)'/%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
      />
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Navire de plaisance pro"
      src="data:image/svg+xml,%3csvg%20height='25'%20viewBox='0%200%2069%2025'%20width='69'%20xmlns='http://www.w3.org/2000/svg'%3e%3cclipPath%20id='a'%3e%3cpath%20d='m0%200h69v25h-69z'/%3e%3c/clipPath%3e%3cg%20clip-path='url(%23a)'%3e%3cpath%20d='m0%200h69v25h-69z'%20fill='transparent'/%3e%3cg%20fill='%233b4559'%3e%3cpath%20d='m542.789%202.775-2.524-2.525h-2.484l1.262%202.525zm0%200'%20transform='translate(-503.203%20-.25)'/%3e%3cpath%20d='m200.4.25h-5.68v2.525h7.58zm0%200'%20transform='translate(-182.207%20-.25)'/%3e%3cpath%20d='m0%20353.267h61.946l1.752-2.525h-63.698zm0%200'%20transform='translate(0%20-328.267)'/%3e%3cpath%20d='m61.385%20233.918h-61.385v5.048h65.492l3.508-5.048zm0%200'%20transform='translate(0%20-218.954)'/%3e%3cpath%20d='m110.971%2078.137h-45.2l-1.888%207.573h54.662zm-41.131%205.049h-2.525v-2.525h2.525zm5.049%200h-2.525v-2.525h2.524zm5.049%200h-2.525v-2.525h2.524zm5.05%200h-2.526v-2.525h2.525zm5.049%200h-2.526v-2.525h2.525zm5.049%200h-2.526v-2.525h2.524zm5.05%200h-2.527v-2.525h2.525zm5.048%200h-2.525v-2.525h2.525zm5.049%200h-2.525v-2.525h2.525zm0%200'%20transform='translate(-59.742%20-73.158)'/%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
